js/blocks/components/fields.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Internal dependencies
+ */
+import simplifiedFields from '../loader/fields';
+import { ControlContainer } from './';
+
+/**
+ * Renders the fields, using their control functions.
+ *
+ * @param {Array}  fields           The fields to render.
+ * @param {Object} parentBlockProps The props to pass to the control function.
+ * @param {Object} parentBlock      The block where the fields are.
+ * @param {number} rowIndex         The index of the repeater row, if this field is in one (optional).
+ * @return {Function} fields The rendered fields.
+ */
+const Fields = ( { fields, parentBlockProps, parentBlock, rowIndex } ) => {
+	return simplifiedFields( fields ).map( ( field ) => {
+
+		/**
+		 * Handles a single control value changing.
+		 *
+		 * Changing a control value inside a block needs to be able to call
+		 * the block's setAttributes property, so that the block can save the value.
+		 * This function is passed to the control so that the control can save the value,
+		 * depending on whether the control is in a repeater row or not.
+		 *
+		 * @param {mixed} newValue The new control value.
+		 */
+		const onChange = ( newValue ) => {
+			const attr = { ...parentBlockProps.attributes };
+			const { setAttributes } = parentBlockProps;
+
+			if ( undefined !== rowIndex ) { // If this is in a repeater row.
```

Comment:
@todo: change order of if/else

Reply:
This is applied in https://github.com/getblocklab/block-lab/pull/298/commits/5abc5644bf97c1123e5864b30a7b6993ca332902